module/os/freebsd/zfs/zfs_vnops_os.c

```diff
@@ -6289,7 +6289,8 @@ zfs_freebsd_copy_file_range(struct vop_copy_file_range_args *ap)
 		goto unlock;
 
 	error = zfs_clone_range(VTOZ(invp), ap->a_inoffp, VTOZ(outvp),
```

Comment:
Why not modify zfs_clone_range to take uint64_t instead of doing this? This is more of a comment on @pjd's file cloning work, but I feel like a 32-bit size_t on a 32-bit machine would pose problems for cloning large files. Doing a type cast here feels like working around a symptom of a problem rather than fixing the actual problem.

Reply:
That might be a problem for 32-bit FreeBSD users, but that is outside the scope if this PR. The PR is fine as is.